### PR TITLE
added CRUD API test for deck

### PR DESCRIPTION
**Pull request by erkita**

### CODE REVIEW

File: server/test/deckTest.js [hunk 1]
```diff
@@ -1,23 +1,104 @@
...
+        res.body.should.have.property("message").eql(invalidNameMessage);
+        done();
+      });
+  });
+});
 
-describe("Decks", function () {
-  it("list first deck GET", (done) => {
+describe("GET Decks", function () {
+  it("GET, get first deck in db", (done) => {
```
Comment (cathylee12) [inline server/test/deckTest.js:101]: I'm curious. Is there any way to retrieve all card decks when get is called? Or are we always return the first one?

File: server/test/deckTest.js [hunk 2]
```diff
@@ -26,7 +107,129 @@ describe("Decks", function () {
...
+      .patch("/decks/" + testDeckId)
+      .send({ name: ">!" })
+      .end((err, res) => {
+        res.should.have.status(404);
+        res.body.should.be.a("object");
+        res.body.should.have.property("message").eql("Valid name required");
+        done();
+      });
+  });
+});
```
Comment (cathylee12) [inline server/test/deckTest.js:220]: What if the input is an empty string? (no white space)

**Review by cathylee12 — COMMENTED** [review]
Testing the total number of decks would be also a good idea

File: server/test/deckTest.js (hunk 2 above)
Comment (erkita) [reply server/test/deckTest.js:220]: tested in line 60

File: server/test/deckTest.js (hunk 1 above)
Comment (erkita) [reply server/test/deckTest.js:101]: Yes but the issue is as mentioned in the edited comment above, number of decks in db will not be static. That's why for now I did the 1st deck in db as I've seen in the [example I shared](https://github.com/mjhea0/node-mocha-chai-tutorial/blob/master/test/test-server.js).

File: server/test/deckTest.js (hunk 2 above)
Comment (cathylee12) [reply server/test/deckTest.js:220]: I thought an empty string for PATCH would be somewhat different, but if not, that's okay.

File: server/test/deckTest.js (hunk 1 above)
Comment (cathylee12) [reply server/test/deckTest.js:101]: My purpose of testing total number was to see if GET properly gets all the decks in db. To do this, it has to be static some way that's why I had to reset Flashcard collection in db in the beginning of testing in my case.. I know this method is not appropriate so I am trying to do it in other way. Your way of testing should be good then

**Review by stefanisindarto — APPROVED** [review]

**Pull request merged**